packages/editor/src/components/post-template/block-theme.js

```diff
@@ -24,21 +24,29 @@ const POPOVER_PROPS = {
 };
 
 export default function BlockThemeControl( { id } ) {
-	const { isTemplateHidden } = useSelect( ( select ) => {
-		const { getRenderingMode } = unlock( select( editorStore ) );
+	const { isTemplateHidden, getPostLinkProps } = useSelect( ( select ) => {
+		const { getRenderingMode, getEditorSettings } = unlock(
+			select( editorStore )
+		);
 		return {
 			isTemplateHidden: getRenderingMode() === 'post-only',
+			getPostLinkProps: getEditorSettings().getPostLinkProps,
 		};
 	}, [] );
 	const { editedRecord: template, hasResolved } = useEntityRecord(
 		'postType',
 		'wp_template',
 		id
 	);
-	const { getEditorSettings } = useSelect( editorStore );
 	const { createSuccessNotice } = useDispatch( noticesStore );
 	const { setRenderingMode } = useDispatch( editorStore );
-
+	const editTemplate = getPostLinkProps
+		? getPostLinkProps( {
+				postId: template.id,
+				postType: 'wp_template',
+				canvas: 'edit',
+		  } )
+		: {};
```

Comment:
Hmm, yeah I see there's quite a lot of use of this now in the editor package.

I'm not sure about the proposed `editPost` naming, as it sounds like it'll carry out the action when called, but it actually returns an object.

`getPostLinkProps` is a bit awkward though too.

An alternative could be to have three settings, `editPost` (for the click event), `getEditPostLink` (for the href) and `goBack`.

Would be good to gather more input on that. I'm still a bit uncomfortable with it being an editor setting TBH.

Reply:
I have a plan for tidying this up a bit. I will put up a separate PR for this if my theory works out in practice.

Reply:
See [this PR](https://github.com/WordPress/gutenberg/pull/57745) for my suggestion on how we can refactor `getPostLinkProps` to work better across the extended use cases.